keras/applications/mobilenet.py

```diff
@@ -0,0 +1,572 @@
+    # Create model.
+    model = Model(inputs, x, name='mobilenet')
+
+    # load weights
+    if weights == 'imagenet':
+        if K.image_data_format() == 'channels_first':
+            raise AttributeError('Weights for Channels Last format are not available')
+
+        if alpha == 1.0:
+            alpha_text = '1_0'
+        elif alpha == 0.75:
+            alpha_text = '7_5'
+        elif alpha == 0.50:
+            alpha_text = '5_0'
+        else:
+            alpha_text = '2_5'
+
+        if include_top:
+            model_name = 'mobilenet_%s_%d_tf.h5' % (alpha_text, rows)
+            weigh_path = BASE_WEIGHT_PATH + model_name
+            weights_path = get_file(model_name,
+                                    weigh_path,
+                                    cache_subdir='models')
+        else:
+            model_name = 'mobilenet_%s_%d_tf_no_top.h5' % (alpha_text, rows)
+            weigh_path = BASE_WEIGHT_PATH + model_name
+            weights_path = get_file(model_name,
+                                    weigh_path,
+                                    cache_subdir='models')
+
+        model.load_weights(weights_path)
+
+    if old_data_format:
+        K.set_image_data_format(old_data_format)
+
+    return model
+
+
+def _conv_block(input, filters, alpha, kernel=(3, 3), strides=(1, 1)):
+    """ Adds an initail convolution layer (with batch normalization and relu6)
+    """
+    channel_axis = 1 if K.image_data_format() == 'channels_first' else -1
+    filters = int(filters * alpha)
+
+    x = Convolution2D(filters, kernel, padding='same', use_bias=False, strides=strides,
+                      name='conv1')(input)
+    x = BatchNormalization(axis=channel_axis, name='conv1_bn')(x)
+    x = Activation(relu6, name='conv1_relu')(x)
+
+    return x
+
+
+def _depthwise_conv_block(input, pointwise_conv_filters, alpha,
+                          depth_multiplier=1, strides=(1, 1), id=1):
+    """ Adds a depthwise convolution block (depthwise conv, batch normalization, relu6,
+        pointwise convolution, batch normalization and relu6)
+    """
+    channel_axis = 1 if K.image_data_format() == 'channels_first' else -1
+    pointwise_conv_filters = int(pointwise_conv_filters * alpha)
+
+    x = DepthwiseConvolution2D(kernel_size=(3, 3), padding='same', depth_multiplier=depth_multiplier,
+                               strides=strides, use_bias=False, name='conv_dw_%d' % id)(input)
+    x = BatchNormalization(axis=channel_axis, name='conv_dw_%d_bn' % id)(x)
+    x = Activation(relu6, name='conv_dw_%d_relu' % id)(x)
+
+    x = Convolution2D(pointwise_conv_filters, (1, 1), padding='same', use_bias=False, strides=(1, 1),
+                      name='conv_pw_%d' % id)(x)
+    x = BatchNormalization(axis=channel_axis, name='conv_pw_%d_bn' % id)(x)
+    x = Activation(relu6, name='conv_pw_%d_relu' % id)(x)
+
+    return x
+
+
+def _create_mobilenet(classes, img_input, include_top, alpha, depth_multiplier, dropout, pooling):
```

Comment:
I found the design here much more practical than the design of other classes in the applications folder implementations. My situations are slightly different so I'm able to just call this internal function. In the applications of other utilities such as [ResNet50](https://github.com/fchollet/keras/blob/master/keras/applications/resnet50.py#L125) I've had to manually copy/paste the code or otherwise rewrite things to actually use it. 

In other words, for models there are very specific and valid reasons why such `_create*()` functions should exist.